http-client-core/src/main/java/io/micronaut/http/client/bind/binders/QueryValueClientArgumentRequestBinder.java

```diff
@@ -0,0 +1,400 @@
+     */
+    public static final String QUERY_VALUE_FORMAT_CSV = "CSV";
+    /**
+     * Values separated with spaces " " similarly to CSV being separated with commas.
+     */
+    public static final String QUERY_VALUE_FORMAT_SSV = "SSV";
+    /**
+     * Values separated with the pipe "|" symbol similarly to CSV being separated with commas.
+     */
+    public static final String QUERY_VALUE_FORMAT_PIPES = "PIPES";
+    /**
+     * Values are repeated with the same parameter name for {@link Iterable}, while {@link Map} and POJO {@link Object}
+     * would be expanded with its property names.
+     * <table border="1">
+     *     <tr> <th><b> Type </b></th>      <th><b> Example value </b></th>                        <th><b> Example representation </b></th> </tr>
+     *     <tr> <td> Iterable &emsp </td>   <td> param=["Mike", "Adam", "Kate"] </td>              <td> "param=Mike&param=Adam&param=Kate </td></tr>
+     *     <tr> <td> Map </td>              <td> param=["name": "Mike", "age": "30"] &emsp </td>   <td> "name=Mike&age=30" </td> </tr>
+     *     <tr> <td> Object </td>           <td> param={name: "Mike", age: 30} </td>               <td> "name=Mike&age=30" </td> </tr>
+     * </table>
+     */
+    public static final String QUERY_VALUE_FORMAT_MULTI = "MULTI";
+    /**
+     * Values are put in the representation with property name for {@link Map} and POJO {@link Object} in square
+     * after the original parameter name.
+     * <table border="1">
+     *     <tr> <th><b> Type </b></th>      <th><b> Example value </b></th>                        <th><b> Example representation </b></th> </tr>
+     *     <tr> <td> Iterable &emsp </td>   <td> param=["Mike", "Adam", "Kate"] </td>              <td> "param[0]=Mike&param[1]=Adam&param[2]=Kate </td></tr>
+     *     <tr> <td> Map </td>              <td> param=["name": "Mike", "age": "30"] &emsp </td>   <td> "param[name]=Mike&param[age]=30" </td> </tr>
+     *     <tr> <td> Object </td>           <td> param={name: "Mike", age: 30} </td>               <td> "param[name]=Mike&param[age]=30" </td> </tr>
+     * </table>
+     */
+    public static final String QUERY_VALUE_FORMAT_DEEP_OBJECT = "DEEP_OBJECT";
+
+    private static final String CSV_DELIMITER = ",";
+    private static final String SSV_DELIMITER = " ";
+    private static final String PIPES_DELIMITER = "|";
+
+    private final ConversionService<?> conversionService;
+
+    public QueryValueClientArgumentRequestBinder(ConversionService<?> conversionService) {
+        this.conversionService = conversionService;
+
+        // Add the converters
+        conversionService.addConverter(Iterable.class, ConvertibleMultiValues.class, new IterableConverter());
+        conversionService.addConverter(Map.class, ConvertibleMultiValues.class, new MapConverter());
+        conversionService.addConverter(Object.class, ConvertibleMultiValues.class, new ObjectConverter());
```

Comment:
It's not appropriate to add the converters here. We have a converter registrar interface you can use which would make them available before this class gets instantiated

Reply:
I wasn't sure that the converters would be useful if someone is not using the micronaut-http or micronaut-client. Should I add them to the converter register anyway?

Reply:
I think they are useful outside of those contexts. For example in configuration, websockets, etc